bsip-0032.md

```diff
@@ -0,0 +1,92 @@
+order to sell 100 BTS at 0.1 USD per BTS, another person (B) then placed a new
+limit order to buy 100 BTS at 0.105 USD per BTS, the two orders will match at
+0.1 USD per BTS, so A will pay 100 BTS and get 10 USD, B will pay 10 USD and
+get 100 BTS.
+
+However, when matching a taker limit order with a maker margin call order,
+the taker price is being used. For example, if trader A's margin call order is
+selling 100 BTS at no less than 0.1 USD per BTS, then trader B placed an order
+that buys 100 BTS at 0.105 USD per BTS, the two order will match at 0.105 USD
+per BTS, so A will pay 100 BTS and get 10.5 USD, B will pay 10.5 USD and get
+100 BTS.
+
+This BSIP proposes a new behavior: always match orders at maker price.
+
+# Motivation
+
+Make the exchange system more user-friendly.
+
+# Rationale
+
+To attract more users, the system should be fair.
+
+It's common sense that orders should be matched at maker price.
+
+# Specifications
+
+There is a parameter in price feed named MSSR, which stands for "maximum short
+squeeze ratio". Maker price of margin call orders is MSSP, which stands for
+"maximum short squeeze price", is calculated as `feed_price / MSSR`.
+Note: `feed_price` here is in terms of debt/collateral, aka "how much debt per
+collateral".
+
+Matching between a limit order and a call order is done in
+`check_call_orders(...)` function of `database` class, price of limit order
+is always used. It need to be changed to use MSSP when the call order is maker.
+
+Currently a black swan event will occur when the call order with least
+collateral ratio is going to be matched below 100% collateral ratio price
+(name it `100CRP`). Because the call order will be matched with incoming limit
+order at limit order's price (taker price),
+which can be higher or lower than 100CRP, so even if MSSP is below 100CRP,
+an incoming taker limit order may or may not trigger a black swan.
+So it makes sense to check if a black swan event will occur every time when a
+limit order is created.
+
+After the behavior changed to always match at maker price, when MSSP is below
+100CRP, an incoming taker limit order will always trigger a black swan event.
+So it makes sense to trigger the black swan event when MSSP is below 100CRP
+rather than waiting for an incoming limit order. That means it's no longer
+needed to check for black swan event when a limit order is created.
+Since checking for black swan event is somehow expensive, we'll gain a side
+benefit on performance with the change.
+
+When triggering a black swan event when MSSP is below 100CRP,
+sometimes the short
+position with least collateral ratio may still have more than 100% collateral
+ratio. In this case, the global settlement price is 100CRP but not the actual
+collateral ratio of the short position with least collateral ratio.
+This is current behavior, it's fair, no need to change.
+
+# Discussion
+
+[to be added if any]
```

Comment:
Suggestion:

> It might seem unfair on the shorter to match at MSSP even if the incoming order specifies a better price. However, in a rationally acting market users will not, in the presence of margin calls, create limit orders below the MSSP. Effectively the new rule doesn't change this situation.